src/main/java/com/google/sps/servlets/UserCardsServlet.java

```diff
@@ -31,77 +26,83 @@
-        Query cardQuery = new Query("Card").setAncestor(KeyFactory.stringToKey(folderKey));
+      Query cardQuery = new Query("Card").setAncestor(KeyFactory.stringToKey(folderKey));
 
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        PreparedQuery results = datastore.prepare(cardQuery);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(cardQuery);
 
-        if (results != null) {
-          for (Entity entity: results.asIterable()) {
-            Card card = Card.EntityToCard(entity);
-            userCards.add(card);
-          }   
-        }
+      if (results != null) {
+        for (Entity entity : results.asIterable()) {
+          String key = KeyFactory.keyToString(entity.getKey());
+          Card card = new Card(entity, key);
+          userCards.add(card);
+        }   
+      }
 
-        response.setContentType("application/json;");
-        response.getWriter().println(new Gson().toJson(userCards));
+      jsonInfo.put("showCreateFormStatus", true);
     }
+    
+    jsonInfo.put("userCards", userCards);
+    response.setContentType("application/json;");
+    response.getWriter().println(new Gson().toJson(jsonInfo));
   }
   
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
     UserService userService = UserServiceFactory.getUserService();
 
     if (userService.isUserLoggedIn()) {
+      String folderKey = request.getParameter("folderKey");
+      String rawText = request.getParameter("rawText");
+      String textTranslated = request.getParameter("translatedText");
+      String imageBlobKey = getImageBlobKey(request);
+
+      Card card = new Card.Builder()
+          .setImageBlobKey(imageBlobKey)
+          .setRawText(rawText)
+          .setTextTranslated(textTranslated)
+          .setParentKey(folderKey)
+          .build();
+  
+      Entity cardEntity = card.createEntity();
 
-        // if blobKey is null, user did not add an image
-        String folderKey = request.getParameter("folderKey");
-        String blobKey = getBlobKeyfromBlobstore(request, "image");
-        String labels = request.getParameter("labels");
-        String fromLang = request.getParameter("fromLang");
-        String toLang = request.getParameter("toLang");
-        String textNotTranslated = request.getParameter("textNotTranslated");
-        String textTranslated = translateText(textNotTranslated, toLang);
-
-        Card card = new Card(blobKey, labels, fromLang, toLang, textNotTranslated, textTranslated);
-        Entity cardEntity = card.createEntity(KeyFactory.stringToKey(folderKey));
-
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        datastore.put(cardEntity);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(cardEntity);
```

Comment:
No response back to browser?

Reply:
Note: we talked about this, so as of right now, we won't be sending a response to the browser